### ISSUE TITLE
Non-existent imports

### ISSUE THREAD

**Issue by joshua-schnabel** [typescript-rtti/typescript-rtti#61]
I have the following TS file. When the transformer translates the file, unknown imports are generated. The statement that generates the error is this import: ` import * as LΦ_1 from "winston/index";` The following error is displayed:`Error [ERR_MODULE_NOT_FOUND]: Cannot find module '.../project/node_modules/winston/index'`.
winston is built like this, so no index.js can be found.
![grafik](https://user-images.githubusercontent.com/5574446/166105962-5fc6a060-4fc4-4a34-8198-87886972f9a2.png)
Is this a bug or am I making a mistake somewhere?

Original File:
```js
import winston, { format, transports, transport as Transport, Logger } from "winston";
import { Format } from "logform";

const { combine, timestamp, label, printf, errors, splat } = format;

// Custom logging format
const customFormat = printf(({ level, message, label, timestamp, stack }) => {
  return `${level}\t ${timestamp} · ${label || "-"} \t· ${message} ${stack || ""}`;
});

const myCustomLevels = {
  levels: {
    alert: 0,
    fatal: 1,
    error: 2,
    warning: 3,
    verbose: 4,
    info: 5,
    debug: 6,
    trace: 7
  },
  colors: {
    alert: "magentaBG white",
    fatal: "redBG white",
    error: "red",
    warning: "yellow",
    verbose: "green",
    info: "blue",
    debug: "cyan",
    trace: "cyanBG black"
  }
};

// Custom combined logging format:
const customCombinedFormat = (module: string): Format =>
  combine(
    errors({ stack: true }),
    format.colorize({ level: true }),
    label({ label: module }),
    timestamp(),
    splat(),
    customFormat
  );

// Custom transports:
const customTransports = (): Transport[] => [new transports.Console()];

// Container to provide different pre-configured loggers
const logContainer = new winston.Container();

winston.addColors(myCustomLevels.colors);

// Default logger for modules:
const getLogger = (module: string): Logger & Record<keyof typeof myCustomLevels["levels"], winston.LeveledLogMethod> => {
  if (!logContainer.has(module)) {
    logContainer.add(module, {
      level: "trace",
      levels: myCustomLevels.levels,
      format: customCombinedFormat(module),
      transports: customTransports()
    });
  }
  return logContainer.get(module) as Logger & Record<keyof typeof myCustomLevels["levels"], winston.LeveledLogMethod>;
};

export default getLogger;
export const defaultLogger =  getLogger("default");
```

Transformed File:
```js
const __RΦ = { m: (k, v) => (t, ...a) => t && Reflect.metadata ? Reflect.metadata(k, v)(t, ...a) : void 0, f: (f, d, n) => (d.forEach(d => d(f)), Object.defineProperty(f, "name", { value: n, writable: false }), f), c: (c, d, dp, dsp, n) => (d.forEach(d => d(c)), dp.forEach(([p, d]) => d(c.prototype, p)), dsp.forEach(([p, d]) => d(c, p)), n ? Object.defineProperty(c, "name", { value: n, writable: false }) : undefined, c), r: (o, a) => (Object.assign(o, a)), a: id => {
        let t = __RΦ.t[id];
        if (t === void 0)
            return void 0;
        if (t.RΦ) {
            let r = t.RΦ;
            delete t.RΦ;
            __RΦ.r(t, r(t));
        }
        else if (t.LΦ) {
            let l = t.LΦ();
            delete t.LΦ;
            __RΦ.t[id] = t = l;
        }
        return t;
    }, t: { [4]: { RΦ: t => ({ TΦ: "~" }) }, [13]: { LΦ: t => String }, [3508]: { LΦ: t => LΦ_0.Format }, [3588]: { RΦ: t => ({ TΦ: "[", e: __RΦ.a(3586) }) }, [3586]: { LΦ: t => LΦ_1.transport }, [3602]: { RΦ: t => ({ TΦ: "&", t: [__RΦ.a(3594), __RΦ.a(3600)] }) }, [3594]: { LΦ: t => LΦ_1.IΦLogger }, [3600]: { RΦ: t => ({ TΦ: "m", t: __RΦ.a(3601), p: [__RΦ.a(3599), __RΦ.a(3598)] }) }, [3601]: { LΦ: t => Object }, [3599]: { RΦ: t => ({ TΦ: "|", t: [__RΦ.a(3566), __RΦ.a(3567), __RΦ.a(3568), __RΦ.a(3569), __RΦ.a(3570), __RΦ.a(3571), __RΦ.a(3572), __RΦ.a(3573)] }) }, [3566]: { LΦ: t => "alert" }, [3567]: { LΦ: t => "fatal" }, [3568]: { LΦ: t => "error" }, [3569]: { LΦ: t => "warning" }, [3570]: { LΦ: t => "verbose" }, [3571]: { LΦ: t => "info" }, [3572]: { LΦ: t => "debug" }, [3573]: { LΦ: t => "trace" }, [3598]: { LΦ: t => LΦ_1.IΦLeveledLogMethod } } };
import winston, { format, transports } from "winston";
const { combine, timestamp, label, printf, errors, splat } = format;
// Custom logging format
const customFormat = printf(__RΦ.f(({ level, message, label, timestamp, stack }) => {
    return `${level}\t ${timestamp} · ${label || "-"} \t· ${message} ${stack || ""}`;
}, [__RΦ.m("rt:p", [{ n: "{ level, message, label, timestamp, stack }", t: () => __RΦ.a(4), v: null }]), __RΦ.m("rt:f", "F>"), __RΦ.m("rt:t", () => __RΦ.a(13))], ""));
const myCustomLevels = {
    levels: {
        alert: 0,
        fatal: 1,
        error: 2,
        warning: 3,
        verbose: 4,
        info: 5,
        debug: 6,
        trace: 7
    },
    colors: {
        alert: "magentaBG white",
        fatal: "redBG white",
        error: "red",
        warning: "yellow",
        verbose: "green",
        info: "blue",
        debug: "cyan",
        trace: "cyanBG black"
    }
};
import * as LΦ_0 from "logform";
// Custom combined logging format:
const customCombinedFormat = __RΦ.f((module) => combine(errors({ stack: true }), format.colorize({ level: true }), label({ label: module }), timestamp(), splat(), customFormat), [__RΦ.m("rt:p", [{ n: "module", t: () => __RΦ.a(13), v: null }]), __RΦ.m("rt:f", "F>"), __RΦ.m("rt:t", () => __RΦ.a(3508))], "customCombinedFormat");
import * as LΦ_1 from "winston/index";
// Custom transports:
const customTransports = __RΦ.f(() => [new transports.Console()], [__RΦ.m("rt:p", []), __RΦ.m("rt:f", "F>"), __RΦ.m("rt:t", () => __RΦ.a(3588))], "customTransports");
// Container to provide different pre-configured loggers
const logContainer = new winston.Container();
winston.addColors(myCustomLevels.colors);
// Default logger for modules:
const getLogger = __RΦ.f((module) => {
    if (!logContainer.has(module)) {
        logContainer.add(module, {
            level: "trace",
            levels: myCustomLevels.levels,
            format: customCombinedFormat(module),
            transports: customTransports()
        });
    }
    return logContainer.get(module);
}, [__RΦ.m("rt:p", [{ n: "module", t: () => __RΦ.a(13), v: null }]), __RΦ.m("rt:f", "F>"), __RΦ.m("rt:t", () => __RΦ.a(3602))], "getLogger");
export default getLogger;
export const defaultLogger = getLogger("default");

```

**Comment by rezonant** [comment]
Looks like a bug triggered due to the TS types not being level with the JS files:

winston's package.json:

```
  "main": "./lib/winston",
  "browser": "./dist/winston",
  "types": "./index.d.ts",
```

Here RTTI is trying to find the most appropriate import to use, it's logic makes sense for modules that are part of your own package but it's rarely appropriate to dig into a node_modules package this way.

We should perform a check for a corresponding .js file in the location it wants to import and if it doesn't exist, fall back to the package's main entry point. (ie "winston") if there's no matching JS file. If it turns out that the desired symbol isn't exported from the entry point, it should result in an "any" type reference at runtime. I think we could emit a warning either at compile time or runtime to aid in troubleshooting any issue that falls out of that.

In the mean time, you can use the `@rtti:skip` JSDoc tag to skip processing a part of your code which is causing problems, if that makes sense for you.

**Comment by rezonant** [comment]
A fix for this issue will arrive in v0.6.1

**Comment by rezonant** [comment]
v0.6.1 is now available, let us know if you run into any other issues!
